FWCore/Framework/interface/es_impl/ReturnArgumentTypes.h

```diff
@@ -11,20 +11,40 @@ namespace edm::eventsetup::impl {
     using argument_type = T;
     using return_type = R;
   };
+  // second argument for WaitingTaskArenaHolder
+  template <typename R, typename T, typename T2>
+  struct ReturnArgumentTypesImpl<R (*)(T const&, T2)> {
```

Comment:
@Dr15Jones I added these additional partial specializations to match the `ESProducerExternalWork`'s "acquire" function signature that has `edm::WaitingTaskWithArenaHolder` in addition to the Record type (i.e. `TReturn acquire(TRecord const&, edm::WaitingTaskWithArenaHolder)`).

I did it first in a generic way by adding additional template argument `T2` (that is then ignored) with the idea of using this same mechanism for the "produce" function, but that didn't work in the end. I'm wondering if it would be better to keep the `T2` template argument, or change this explicitly use `edm::WaitingTaskWithArenaHolder`?
```suggestion
  template <typename R, typename T>
  struct ReturnArgumentTypesImpl<R (*)(T const&, edm::WaitingTaskWithArenaHolder)> {
```

Reply:
We decided that explicit `WaitingTaskWithArenaHolder` argument would be better